SparkPost/2web2ui
# SA-1586: Save Report modal

### What Changed
 - Modal appears when clicking "Save Changes"
 - Shows form to update and save report

### How To Test
 - Go to Report builder
 - Select or save a new report
 - Change the metrics
 - Click "Save Report"
 - Verify that the metrics and dates are listed for the report
 - Click save
 - Verify API request succeeds

### To Do
- [ ] Update list of reports in the typeahead list
- [ ] Update query param to include the id for the report
